src/lib/helpers/CartHelper.ts

```diff
@@ -1,41 +1,40 @@
-import { CartItem, extractedProductsList } from "../types/Types";
+import { CartItem, extractedProductsList } from '../types/Types';
 
 export const getUserCartFromStorage = () => {
-  const data = localStorage.getItem("userCart");
-  if (data) {
-    return JSON.parse(data);
-  } else {
-    return null;
-  }
+    const data = localStorage.getItem('userCart');
+    if (data) {
+        return JSON.parse(data);
+    } else {
+        return null;
+    }
 };
 
 export const setUserCartToStorage = (cart: CartItem[] | any) => {
-  return localStorage.setItem("userCart", JSON.stringify(cart));
+    return localStorage.setItem('userCart', JSON.stringify(cart));
 };
-export const mergeStockArray = (
-  initArray: extractedProductsList[],
-  id: number,
-  quantity: number
-) => {
-  const copyInitArray = structuredClone(initArray);
-  const findedProductIndex = copyInitArray.findIndex((el) => el.id == id);
-  if (findedProductIndex == -1) {
-    copyInitArray.push({ id, quantity });
-  } else {
-    copyInitArray[findedProductIndex].quantity = quantity;
-  }
-  return copyInitArray;
+
+export const deleteUserCartToStorage = () => {
+    localStorage.removeItem('userCart');
 };
 
-export const deleteItemFromCart = (
-  initArray: extractedProductsList[],
-  id: number
-) => {
-  const copyInitArray = structuredClone(initArray);
-  const findedProductIndex = copyInitArray.findIndex((el) => el.id == id);
-  if (findedProductIndex != -1) {
-    copyInitArray.splice(findedProductIndex,1);
+export const mergeStockArray = (initArray: extractedProductsList[], id: number, quantity: number) => {
+    console.log(initArray);
+    const copyInitArray = structuredClone(initArray);
+    const findedProductIndex = copyInitArray.findIndex((el) => el.id == id);
+    if (findedProductIndex == -1) {
+        copyInitArray.push({ id, quantity });
+    } else {
+        copyInitArray[findedProductIndex].quantity = quantity;
+    }
     return copyInitArray;
-  }
-  return initArray;
+};
+
+export const deleteItemFromCart = (initArray: extractedProductsList[], id: number) => {
+    const copyInitArray = structuredClone(initArray);
+    const findedProductIndex = copyInitArray.findIndex((el) => el.id == id);
+    if (findedProductIndex != -1) {
+        copyInitArray.splice(findedProductIndex, 1);
+        return copyInitArray;
```

Comment:
please, rewrite with slice.
it will copy and return modified array

Reply:
+